src/stp/new_skills/Rotate.cpp

```diff
@@ -3,45 +3,48 @@
 //
 
 #include "include/roboteam_ai/stp/new_skills/Rotate.h"
+
 #include <roboteam_utils/Print.h>
 
 namespace rtt::ai::stp {
 
-void Rotate::onInitialize() noexcept { }
+void Rotate::onInitialize() noexcept {}
 
-Status Rotate::onUpdate(const rtt::ai::stp::SkillInfo &info) noexcept {
-    float angle = info.getAngle();
-    int dribblerSpeed = info.getDribblerSpeed();
+Status Rotate::onUpdate(const StpInfo &info) noexcept {
+    RTT_WARNING("UPDATING ROTATE")
+    float targetAngle = info.getAngle();
+    int targetDribblerSpeed = info.getDribblerSpeed();
 
     // Check if angle is in range
-    if (angle < -M_PI || angle > M_PI) {
+    if (targetAngle < -M_PI || targetAngle > M_PI) {
         RTT_ERROR("Rotation angle not within acceptable range")
         return Status::Failure;
     }
 
     // Check if dribbler speed is in range
-    if (dribblerSpeed < 0 || dribblerSpeed > 31) {
+    if (targetDribblerSpeed < 0 || targetDribblerSpeed > 31) {
         RTT_ERROR("Dribbler speed not within acceptable range")
         return Status::Failure;
     }
```

Comment:
Again, skills can't really fail. Consider clamping. Consider also the mapping from percentage in this case